File: archives/python/1260.py
import sys
input = sys.stdin.readline

from collections import deque
def dfs(graph, init, visited):
	visited[init] = True
	print(init, end = " ")
	for i in graph[init]:
		if not visited[i]:
			dfs(graph, i, visited)

def bfs(graph, init, visited):
	queue = deque([init])
	visited[init] = True
	while queue:
		v = queue.popleft()
		print(v, end = " ")
		for i in graph[v]:
			if not visited[i]:
				queue.append(i)
				visited[i] = True

N, M, V = map(int, input().split())
graph = [[] for _ in range(N+1)]
for _ in range(M):
	a, b = map(int, input().split())
	graph[a].append(b)
	graph[b].append(a)

for row in graph:
	row.sort()

visited = [False] * (N+1)
dfs(graph, V, visited)
print()
visited = [False] * (N+1)
bfs(graph, V, visited)

File: archives/cpp/1260.cpp
#include <bits/stdc++.h>
#define FOR(i, start, end) for (int i = start; i < end; i++)
#define endl "\n"
#define fastio ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
// #pragma GCC optimize("Ofast")
using namespace std;
using ll = long long;
//-------------------------------------

int N, M, V;
vector<int> graph[1001];
bool visited[1001];
queue<int> q;

void dfs(int node) {
    visited[node] = true;
    cout << node << ' ';
    for (auto i: graph[node]) {
        if (!visited[i]) {
            dfs(i);
        }
    }
}

void bfs(int start) {
    q.push(start);
    visited[start] = true;
    while (!q.empty()) {
        int node = q.front();
        q.pop();
        cout << node << ' ';
        for (auto i: graph[node]) {
            if (!visited[i]) {
                q.push(i);
                visited[i] = true;
            }
        }
    }
}

int main() {
    fastio

    cin >> N >> M >> V;
    FOR(i, 0, M) {
        int a, b; cin >> a >> b;
        graph[a].push_back(b);
        graph[b].push_back(a);
    }

    FOR(i, 1, N+1) {
        sort(graph[i].begin(), graph[i].end());
    }

    dfs(V);
    cout << endl;
    memset(visited, false, sizeof(visited));
    bfs(V);

    return 0;
}